locallib.php

```diff
@@ -1543,7 +1511,45 @@ function moodleoverflow_print_post($post, $discussion, $moodleoverflow, $cm, $co
     return $renderer->render_post($mustachedata);
 }
 
+/**
+ * Check if the limited answer setting is currently disabling answers.
+ * @param stdClass $limitedanswersetting Two Unix timestamp wrapped in a stdClass, upper and lower label for answering.
+ * @return bool
+ */
+function is_currently_time_limited($limitedanswersetting) {
+    return ($limitedanswersetting->la_starttime != 0 && $limitedanswersetting->la_starttime > time())
+    || ($limitedanswersetting->la_endtime != 0 && $limitedanswersetting->la_endtime < time());
+}
 
+/**
+ * Renders the answer action in a post.
+ * @param String $textmuted
+ * @param array $commands
+ * @param stdClass $str
+ * @param String $infolimited
+ * @param String $role
+ * @return void
+ * @throws coding_exception
+ */
+function render_limited_answer($textmuted, &$commands, $str, $infolimited, $role) {
```

Comment:
textmuted, commands and str should be described as the use is not directly understandable